docs/guides/python/podcast-transcription.mdx

```diff
@@ -0,0 +1,497 @@
+```
+
+Next, let's install our base dependencies, then add the `openai-whisper` library as an optional dependency.
+
+```bash
+# Install the base dependencies
+uv sync
+# Add OpenAI whisper dependency
+uv add openai-whisper --optional ml
+```
+
+<Note>
+  We add the extra dependencies to the 'ml' optional dependencies to keep them
+  separate since they can be quite large. This lets us just install them in the
+  containers that need them.
+</Note>
+
+We'll organize our project structure like so:
+
+```text
++--src/
+|  +--__init__.py
+|  +--resources.py
+|  +--services/
+|      +--__init__.py
+|      +--main.py
+|  +--jobs/
+|      +--__init__.py
+|      +--api.pypy
++--nitric.yaml
++--docker/
+|  +-- transcribe.dockerfile
+|  +-- transcribe.dockerignore
+|  +-- python.dockerfile
+|  +-- python.dockerignore
++--.gitignore
++--.python-version
++--pyproject.toml
++--README.md
+```
+
+## Define our resources
+
+We'll start by creating a file to define our Nitric resources. For this project we'll need an API, Batch Job, and two buckets, one for the audio files to be transcribed and one for the resulting transcripts. The API will interface with the buckets, while the Batch Job will handle the transcription.
+
+```python title:src/resources.py
+from nitric.resources import job, bucket, api
+
+main_api = api("main")
+
+transcribe_job = job("transcribe")
+
+podcast_bucket = bucket("podcasts")
+transcript_bucket = bucket("transcripts")
+```
+
+## Add our transcription
+
+Now that we have defined resources, we can import our API and add some routes to access the buckets. Start by importing the resources and adding permissions to the resources.
+
+```python title:src/services/api.py
+from src.resources import main_api, transcript_bucket, podcast_bucket, transcribe_job
+from nitric.application import Nitric
+from nitric.resources import BucketNotificationContext
+from nitric.context import HttpContext
+
+readable_transcript_bucket = transcript_bucket.allow("read")
+submittable_transcribe_job = transcribe_job.allow("submit")
+
+Nitric.run()
+```
+
+We'll then write a route for getting a file from the transcription bucket. These will get a signed download url and redirect the user to this url for downloading the text content.
+
+```python title:src/services/api.py
+# !collapse(1:7) collapsed
+from src.resources import main_api, transcript_bucket, podcast_bucket, transcribe_job
+from nitric.application import Nitric
+from nitric.resources import BucketNotificationContext
+from nitric.context import HttpContext
+
+readable_transcript_bucket = transcript_bucket.allow("read")
+submittable_transcribe_job = transcribe_job.allow("submit")
+
+@main_api.get("/podcast/:name")
+async def get_podcast(ctx: HttpContext):
+    name = ctx.req.params['name']
+
+    download_url = await readable_transcript_bucket.file(f"{name}-transcript.txt").download_url()
+
+    ctx.res.headers["Location"] = download_url
+    ctx.res.status = 303
+
+    return ctx
+
+Nitric.run()
+```
```

Comment:
Working through the guide, the doubling up of code blocks was a bit confusing in some cases. Is there a way we could merge them and integrate the explanation into the code block itself?